src/synthesizer/load_data/load_eagle.py

```diff
@@ -0,0 +1,932 @@
+    individual chunks
+
+    Args:
+        fileloc (string):
+            eagle data file location
+        tag (string):
+            snapshot tag to load
+        numThreads (int)
+            number of threads to use
+        chunk (int)
+            file number to process
+        tot_chunks (int)
+            total number of files to process
+        verbose (bool)
+            Are we talking?
+
+    Returns:
+        a dictionary of Galaxy objects with stars and gas components
+    """
+
+    # get the redshift from the given eagle tag
+    zed = float(tag[5:].replace("p", "."))
+    if chunk > tot_chunks:
+        exceptions.InconsistentArguments(
+            "Value specified by 'chunk' should be lower"
+            "than the total chunks (tot_chunks)"
+        )
+
+    with h5py.File(
+        f"{fileloc}/groups_{tag}/eagle_subfind_tab_{tag}.{chunk}.hdf5", "r"
+    ) as hf:
+        sgrpno = np.array(hf.get("/Subhalo/SubGroupNumber"))
+        grpno = np.array(hf.get("/Subhalo/GroupNumber"))
+
+    if grpno.dtype == object:
+        return []
+
+    # Get required star particle properties
+    s_sgrpno = read_array(
+        "PARTDATA",
+        fileloc,
+        tag,
+        "/PartType4/SubGroupNumber",
+        numThreads=numThreads,
+        verbose=verbose,
+    )
+    s_grpno = read_array(
+        "PARTDATA",
+        fileloc,
+        tag,
+        "/PartType4/GroupNumber",
+        numThreads=numThreads,
+    )
+    # Create mask for particles in the current chunk
+    ok = np.in1d(s_sgrpno, sgrpno) * np.in1d(s_grpno, grpno)
+    s_sgrpno = s_sgrpno[ok]
+    s_grpno = s_grpno[ok]
+
+    s_coords = read_array(
+        "PARTDATA",
+        fileloc,
+        tag,
+        "/PartType4/Coordinates",
+        noH=True,
+        physicalUnits=True,
+        numThreads=numThreads,
+        verbose=verbose,
+    )[ok]  # physical Mpc
+    s_imasses = (
+        read_array(
+            "PARTDATA",
+            fileloc,
+            tag,
+            "/PartType4/InitialMass",
+            noH=True,
+            physicalUnits=True,
+            numThreads=numThreads,
+            verbose=verbose,
+        )[ok]
+        * 1e10
+    )  #  Msun
+    s_masses = (
+        read_array(
+            "PARTDATA",
+            fileloc,
+            tag,
+            "/PartType4/Mass",
+            noH=True,
+            physicalUnits=True,
+            numThreads=numThreads,
+            verbose=verbose,
+        )[ok]
+        * 1e10
+    )  #  Msun
+    s_ages = read_array(
+        "PARTDATA",
+        fileloc,
+        tag,
+        "/PartType4/StellarFormationTime",
+        numThreads=numThreads,
+        verbose=verbose,
+    )[ok]
+    s_ages = get_age(s_ages, zed, numThreads=numThreads)  # Gyr
+    s_Zsmooth = read_array(
+        "PARTDATA",
+        fileloc,
+        tag,
+        "/PartType4/SmoothedMetallicity",
+        numThreads=numThreads,
+        verbose=verbose,
+    )[ok]
+    # s_oxygen = read_array(
```

Comment:
Remove commented code (if it might be needed there's no harm loading these variables always, if it's never going to be needed there's no point including it)

Reply:
Same with mentions below